test/externalTests/external_test.py

```diff
@@ -0,0 +1,368 @@
+        else:
+            os.chdir(test_dir.parent)
+            run_cmd(f"git clone --depth 1 {repo_url} -b {ref} {test_dir.resolve()}")
+            if not test_dir.exists():
+                raise RuntimeError("Git clone failed.")
+            os.chdir(test_dir)
+
+        if (test_dir / ".gitmodules").exists():
+            run_cmd("git submodule update --init")
+
+        commit_hash = subprocess.getoutput("git rev-parse HEAD")
+        print(f"Current commit hash: {commit_hash}")
+
+    @staticmethod
+    def parse_solc_version(solc_version_string):
+        solc_version_match = re.search(
+            SOLC_FULL_VERSION_REGEX, solc_version_string)
+        if solc_version_match:
+            solc_full_version = solc_version_match.group(1)
+        else:
+            raise RuntimeError(
+                f"Solc version could not be found in: {solc_version_string}")
+        return solc_full_version
+
+    @staticmethod
+    def get_solc_short_version(solc_full_version):
+        solc_short_version_match = re.search(
+            SOLC_SHORT_VERSION_REGEX, solc_full_version)
+        if solc_short_version_match:
+            solc_short_version = solc_short_version_match.group(1)
+        else:
+            raise RuntimeError(
+                f"Error extracting short version string from: {solc_full_version}")
+        return solc_short_version
+
+    def setup_solc(self, test_dir: Path) -> (str, str):
+        sc_config = self.config.solc
+
+        if sc_config.binary_type == "solcjs":
+            # FIXME: install_dir is assumed to be relative path
+            solc_dir = test_dir.parent / sc_config.install_dir
+            solc_bin = solc_dir / "dist/solc.js"
+
+            print("Setting up solc-js...")
+            if sc_config.solcjs_src_dir == "":
+                self.download_project(
+                    solc_dir, "https://github.com/ethereum/solc-js.git", "branch", sc_config.branch)
+            else:
+                print(
+                    f"Using local solc-js from {sc_config.solcjs_src_dir}...")
+                copytree(sc_config.solcjs_src_dir, solc_dir)
+                rmtree(solc_dir / "dist")
+                rmtree(solc_dir / "node_modules")
+            os.chdir(solc_dir)
+            run_cmd("npm install")
+            run_cmd("npm run build")
+            copyfile(sc_config.binary_path, solc_dir / "dist/soljson.js")
+            solc_version_output = subprocess.getoutput(
+                f"node {solc_bin} --version")
+        else:
+            print("Setting up solc...")
+            solc_version_output = subprocess.getoutput(
+                f"{sc_config.binary_path} --version | tail -n 1")
+
+        return self.parse_solc_version(solc_version_output)
+
+    def store_benchmark_report(self):
+        # TODO
+        raise NotImplementedError()
+
+    def prepare_node_env(self, test_dir: Path):
+        if which("node") is None:
+            # TODO: install nodejs
```

Comment:
Wait, this script should definitely not be installing node on its own :) It's the job of CI to do that. We're using `cimg/node` image so you can assume that node is always there for tests that need it.

Reply:
No, I should have removed this.